unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/AbstractImmutableByteHashSetTestCase.java

```diff
@@ -228,4 +228,14 @@ public void toImmutable()
         Assert.assertEquals(1, this.newWith((byte) 1).toImmutable().size());
         Assert.assertEquals(3, this.newWith((byte) 1, (byte) 2, (byte) 3).toImmutable().size());
     }
+
+    @Test
+    public void union()
```

Comment:
I think we should add tests for both sets being empty, and for either being empty.

Reply:
Added tests for 1. both sets being empty. 2. left (set1) is empty 3. right (set2) is empty.
Lines: 250-263.